### PR TITLE
Resizeable SplitViewPane

### LINKED ISSUE #3763: Can SplitView have a GridSplitter or similar so that user can resize the left and right parts?
**Is your feature request related to a problem? Please describe.**
I need a SplitView to have a GridSplitter like control in it so that user can resize the left and right panel.

### PR DESCRIPTION
**Pull request by timunie**
**Describe the changes you have made to improve this project**

- [x] Add a Thumb to Resize the Open Pane
- [x] Add a `Style` for the `Thumb`
- [x] Add a `bool` to indicate if Pane is Resizeable
- [x] Add `MinimumOpenPaneLength` and `MaximumOpenPaneLength`

**Closed Issues**

Closes #3763

### CODE REVIEW

**Comment by timunie** [comment]
Here is a Preview: 

SplitViewResizable

File: src/MahApps.Metro/Themes/HamburgerMenuTemplate.xaml [hunk 1]
```diff
@@ -384,7 +384,11 @@
                                 DisplayMode="{TemplateBinding DisplayMode}"
                                 IsPaneOpen="{Binding IsPaneOpen, Mode=TwoWay, RelativeSource={RelativeSource TemplatedParent}}"
                                 IsTabStop="False"
-                                OpenPaneLength="{TemplateBinding OpenPaneLength}"
+                                OpenPaneLength="{Binding RelativeSource={RelativeSource Mode=TemplatedParent}, Path=OpenPaneLength, Mode=TwoWay}"
+                                MinimumOpenPaneLength="{TemplateBinding MinimumOpenPaneLength}"
+                                MaximumOpenPaneLength="{TemplateBinding MaximumOpenPaneLength}"
+                                CanResizeOpenPane="{TemplateBinding CanResizeOpenPane}"
+                                ResizeThumbStyle="{TemplateBinding ResizeThumbStyle}"
```
Comment (punker76) [inline src/MahApps.Metro/Themes/HamburgerMenuTemplate.xaml:391]: These properties must be implemented in HamburgerMenu too.

Comment (timunie) [reply src/MahApps.Metro/Themes/HamburgerMenuTemplate.xaml]: It should work now. The issue was that I used `TemplateBinding`. Now I use `RelativeSource TemplatedParent` instead. The Properties are in the separate file `HamburgerMenu.Properties.cs`

**Comment by timunie** [comment]
> Please look at the missing properties at HamburgerMenu.
> 
> I also got this while I'm dragging
> 
> ```
> System.Windows.Data Error: 8 : Cannot save value from target back to source. BindingExpression:Path=OpenPaneLength; DataItem='HamburgerMenu' (Name='HamburgerMenuControl'); target element is 'SplitView' (Name='MainSplitView'); target property is 'OpenPaneLength' (type 'Double') ArgumentException:'System.ArgumentException: '-503' is not a valid value for the 'System.Windows.FrameworkElement.Width' property on a Setter.
>    at System.Windows.Setter.Seal()
>    at System.Windows.SetterBaseCollection.Seal()
>    at System.Windows.Style.Seal()
>    at MahApps.Metro.Controls.HamburgerMenu.ChangeItemFocusVisualStyle() in d:\projects\git\MahApps.Metro\src\MahApps.Metro\Controls\HamburgerMenu\HamburgerMenu.Properties.cs:line 463
>    at MahApps.Metro.Controls.HamburgerMenu.OpenPaneLengthPropertyChangedCallback(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args) in d:\projects\git\MahApps.Metro\src\MahApps.Metro\Controls\HamburgerMenu\HamburgerMenu.Properties.cs:line 26
>    at System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
>    at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
>    at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
>    at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
>    at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)
>    at MS.Internal.Data.PropertyPathWorker.SetValue(Object item, Object value)
>    at System.Windows.Data.BindingExpression.UpdateSource(Object value)'
> ```

I don't get this error anymore. But you should double check anyway 😄 

Have a great weekend
Tim

File: src/MahApps.Metro/Themes/HamburgerMenuTemplate.xaml (hunk 1 above)
Comment (punker76) [reply src/MahApps.Metro/Themes/HamburgerMenuTemplate.xaml]: 👍  Can you make the `CanResizeOpenPane` default value to `false`? IMO this should be the default expectation. Thx.

Comment (timunie) [reply src/MahApps.Metro/Themes/HamburgerMenuTemplate.xaml]: Yes, ok. I have changed this and added Options to the Demo instead.

**Comment by punker76** [comment]
@timunie your changes have been merged, thanks for your contribution 👍

**Pull request merged**